File: pmd/python/testes/sampletest.py
import unittest

class SampleTest(unittest.TestCase):

  def setUp(self): #equivalente ao BeforeAll do JUnit
    self.listaVazia = []
  
  def tearDown(self): #equivalente ao AfterAll do JUnit
    del self.listaVazia
  
  def test_comportamento(self):
    self.assertEqual(
      len(self.listaVazia), 0, "A lista vazia deve ter 0 elementos")

  def test_IndexError(self):
    with self.assertRaises(IndexError):
      x = self.listaVazia[0]


if __name__ == '__main__':
  unittest.main()

File: pmd/java/testes/SampleTest.java
package testes;

import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

public class SampleTest {

 private java.util.List<Object> listaVazia;

 //Este método será chamado antes de cada método de teste.
 @BeforeAll
 protected void inicializar() {
	 listaVazia = new java.util.ArrayList<Object>();
 } 

 //Após cada método de teste, este método é invocado para limpar o lixo.
 @AfterAll
 protected void terminou() {
	 listaVazia = null;
 }

 //Todo método com a anotação @Test será executado pelo JUnit.
 //O método comportamento faz uma chamada ao método assertEquals 
 //(mensagem, valor esperado, valor atual) que verifica se o resultado 
 //de "listaVazia.size()" é igual a 0 (zero) que é o valor esperado, 
 //caso contrário emite uma mensagem.
 @Test
 public void comportamento() {
   Assertions.assertEquals(
    0, listaVazia.size(), "A lista vazia deve ter 0 elementos");
 }

 //Este método testa a exceção. Ela deve ser lançada, pois não existe
 //nenhum item na lista. Caso contrário a implementação tem problemas.
 @Test
 public void lancarException() {
  IndexOutOfBoundsException e = Assertions.assertThrows(
    IndexOutOfBoundsException.class,
    () -> { listaVazia.get(0); },
    "Era esperada uma IndexOutOfBoundsException, mas não foi lançada."	 
   );
 }
}